src/asm/lexer.c

```diff
@@ -921,6 +921,40 @@ void lexer_DumpStringExpansions(void)
 	free(stack);
 }
 
+/* Discards an block comment */
+static void discardBlockComment(void)
+{
+	dbgPrint("Discarding block comment\n");
+	bool block_ended = false;
+
+	for (;;) {
+		switch(nextChar()) {
+			case EOF:
+				error("Unterminated block comment\n");
+				return;	
+			case '/':
+				if (peek(0) == '*') {
+					warning(WARNING_NESTED_COMMENT,
+						"/* in block comment\n");
+					continue;
+				}
+				/* fallthrough */
+			case '*':
+				if (peek(0) == '/') {
+					block_ended = true;
```

Comment:
To remove the need for `block_ended`, it'd be simpler to `shiftChars(1)` here, and then return.